lib/Parse/ParseExpr.cpp

```diff
@@ -413,6 +413,7 @@ ParserResult<Expr> Parser::parseExprSequenceElement(Diag<> message,
       diagnose(Tok.getLoc(), diag::expected_await_not_async)
         .fixItReplace(Tok.getLoc(), "await");
     }
+    TokReceiver->registerTokenKindChange(Tok.getLoc(), tok::contextual_keyword);
```

Comment:
Any issue with `Tok.setKind(tok::contextual_keyword)`?

Reply:
No, I didn’t know `Tok.setKind` was preferred. What’s the difference between the two, incidentally? Also: When should you use one or the other?

Reply:
`TokReceiver->registerTokenKindChange()` only affects `TokenRecorder` (i.e. `SourceFile.getAllTokens()`), whereas `Tok.setKind()` affects libSyntax/SwiftSyntax parsing results _too_.
I don't think there's a reason to introduce a new usage of `registerTokenKindChange()` at this point.